packages/office-ui-fabric-react/src/components/Rating/Rating.tsx

```diff
@@ -120,8 +122,21 @@ export class Rating extends BaseComponent<IRatingProps, IRatingState> {
     );
   }
 
-  private _getIcon(): JSX.Element {
-    return <i className={ css('ms-Icon', this.props.icon || 'ms-Icon--FavoriteStarFill') } />;
+  private _getIcon(starSize?: number): JSX.Element {
+    if (starSize && starSize < 1) {
+      return (
+        <span className={ css('ms-rating-halfStarContainer', styles.halfStarContainer, 'ms-Icon', this.props.icon || 'ms-Icon--FavoriteStarFill') } >
+          <i className={ css('ms-Rating-star', styles.star,
+            'is-selected ' + styles.starIsSelected,
+            { ['is-disabled ' + styles.starIsDisabled]: this.props.disabled },
+            'ms-rating-halfStar', styles.halfStar, 'ms-Icon',
+            this.props.icon || 'ms-Icon--FavoriteStarFill') }
+            style={ { width: starSize * 100 + '%' } } />
```

Comment:
@mikewheaton We thought about doing just that but this was causing the thing to get more complex - since the actual rating is controlled by radio button, this would mean a change in behavior - earlier we just used to floor the value. So 3.8 would be treated as 3 earlier and will be 4 now. This could result in break in the functionality of previous clients.
For rounding off to nearest 0.5 thing, the consumer can round off the value on its side in case he wants to - this will keep things clean and transparent.
For our consumption purposes, we need the exact same behavior that you pointed out and we plan to implement it on the parent component which consumes Rating component